docs/concepts/juttle_tutorial.md

```diff
@@ -2,27 +2,532 @@
+
+```
+npm install -g juttle
+```
+
+and that you have placed the dataset in `/tmp/data.json` location.
+
+For examples that generate visualizations in the browser, you need to have the outrigger package installed:
+
+```
+npm install -g outrigger
+```
+
+and the outriggerd daemon running.
+
+To run a given example via outrigger client, save the Juttle code to a file, then execute:
+
+```
+outrigger-client browser --path '/tmp/example.juttle'
+```
+
+To run a given example via Juttle CLI (this only works for examples that don't attempt to output to charts), you can reference the saved file:
+
+```
+juttle '/tmp/example.juttle'
+```
+
+or simply launch the `juttle` shell, enter `<` for multiline mode, then paste the code followed by `.` on its own line, enter to execute.
+
+We encourage you to run the provided examples and further experiment by modifying their Juttle code, but also provide screenshots of the expected output for cheaters.
+
+## Dataset
+
+This tutorial uses a subset of real data from the
+[Github Archive](https://www.githubarchive.org/). A small dataset of the same format is provided [here](../examples/datasets/github_data.json) and all Juttle programs in this tutorial will work with it, but for more interesting data, download the 30MB version from [here](XXX where to host). Rather than describing the dataset any further, we will use Juttle to explore it.
+
+## Reading the data
+
+The `read` source is used to feed the data into the Juttle flowgraph. We will start with `read file` (where "file" is an adapter to the local filesystem); another option would be to `read http` to access the file remotely via http-accessible endpoint, but we recommend against that for the 30MB dataset to save time and bandwidth.
+
+To see what the data looks like, let's read it in and display only the first few points:
+
+```juttle
+read file -file '/tmp/data.json'
+| head 5
+| view table
+```
+
+The `view table` sink is the default one, and can be omitted. Let's look at the last few points:
+
+```juttle
+read file -file '/tmp/data.json'
+| tail 5
+```
+
+The rest of the examples will only specify the sink if other than table view is desired.
+
+As we start analyzing the data, you will see that doing certain operations on a large file can get slow. At that point we will reload the data into ElasticSearch and use `read elastic` adapter in the Juttle programs, to take advantage of ES optimizations.
+
+We will also try enriching the data with metadata from a CSV [file](../examples/datasets/github_metadata.csv), but not right away.
+
+## Exploring the data
+
+First, let's look at a few events using the `read file` source. We already used `head` and `tail` processors to limit the data we output, now we will try filtering by timestamp. Juttle defaults to looking for the timestamp in the field `time` unless `-timeField <fieldname>` option is used.
+
+XXX/dmehra this is currently unsupported:
```

Comment:
@dmehra I can add this feature to the file adapter but would want to wait for this other PR to get in before fiddling in that are of the code: https://github.com/juttle/juttle/pull/11